includes/stringbuilder-perf-note.md

```diff
@@ -0,0 +1,25 @@
+Using character-based indexing with the <xref:System.Text.StringBuilder.Chars%2A> property can be extremely slow under the following conditions:
+
+- The <xref:System.Text.StringBuilder> instance is large (for example, it consists of several tens of thousands of characters).
+- The <xref:System.Text.StringBuilder> is "chunky." That is, repeated calls to methods such as <xref:System.Text.StringBuilder.Append%2A?displayProperty=nameWithType> have automatically expanded the object's <xref:System.Text.StringBuilder.Capacity%2A?displayProperty=nameWithType> property and allocated new chunks of memory to it.
+
+Performance is severely impacted because each character access walks the entire linked list of chunks to find the correct buffer to index into.
+
+> [!NOTE]
+>  Even for a large "chunky" <xref:System.Text.StringBuilder> object, using the <xref:System.Text.StringBuilder.Chars%2A>property for index-based access to one or a small number of characters has a negligible performance impact; typically, it is an **0(n)** operation. The significant performance impact occurs when iterating the characters in the  <xref:System.Text.StringBuilder> object, which is an **O(n^2)** operation. 
```

Comment:
should be "interacting with the characters" instead of "interacting the characters".

Reply:
nitpick - there's a double space before the file StringBuilder xref. 🙂

Reply:
@jzeferino the verb is actually iterate instead of interact. However, I think that the correct would be to say iterating through the characters. 

e.g. https://english.stackexchange.com/questions/23023/iterate-vs-iterate-over

What do you think @rpetrusha?